### PR TITLE
CI: Run build-check and lint in parallel

### PR DESCRIPTION
**Pull request by toddlerer**
Running CI in parallel makes it faster and easier to see.
Separated build check and lint and changed a few options.

@mtshiba

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -22,32 +22,37 @@ jobs:
...
-      - name: Build
-        run: |
-          rustup update stable
-          cargo build --verbose
-      - name: Run tests
-        # HACK: The cause is unknown, but on windows, the exit code is 1 even if tests are successful.
-        run: cargo test --features large_thread --features pre-commit --verbose --all
-      - uses: actions-rs/cargo@v1
+      - run: rustup update stable
+      - run: cargo build --all --all-targets --verbose --release
```
Comment (toddlerer) [inline .github/workflows/rust.yml:46]: This may be excessive.

Comment (mtshiba) [reply .github/workflows/rust.yml:46]: Hmm, it might be better to have a separate issue on this.

**Comment by mtshiba** [comment]
Thanks!

File: .github/workflows/rust.yml (hunk 1 above)
Comment (mtshiba) [reply .github/workflows/rust.yml:46]: Is the `--release` flag necessary? I think it would be faster without it.

Comment (toddlerer) [reply .github/workflows/rust.yml:46]: I think this level of CI time is acceptable, so I will assume it is not so important now.

**Pull request merged**